Invite Users module it's not working as expected

<!---


<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

Terraform v1.0.0
on darwin_amd64
+ provider registry.terraform.io/ibm-cloud/ibm v1.23.0


### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_XXXXX

### Terraform Configuration Files

<!--- provider "ibm" {
}

module "user_invite" {
  source = "terraform-ibm-modules/iam/ibm//modules/users"

  user_email_addresses               = var.user_email_addresses
  access_groups                      = var.access_groups
}-->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please share a link to the ZIP file.
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output
module.user_invite.ibm_iam_user_invite.invite_user: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with module.user_invite.ibm_iam_user_invite.invite_user,
│   on .terraform/modules/user_invite/modules/users/main.tf line 6, in resource "ibm_iam_user_invite" "invite_user":
│    6: resource "ibm_iam_user_invite" "invite_user" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-ibm_v1.23.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 123 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm.getInfraPermissions(0xc00042f680, 0x369aaa0, 0xc000bc4000, 0xc0004aee70, 0xc000b356c0, 0xc000b39300)
        /Users/harinireddy/go/src/github.com/IBM-Cloud/terraform-provider-ibm/ibm/resource_ibm_iam_user_invite.go:851 +0x60f
github.com/IBM-Cloud/terraform-provider-ibm/ibm.resourceIBMIAMInviteUsers(0xc00042f680, 0x369aaa0, 0xc000bc4000, 0x0, 0xffffffffffffffff)
        /Users/harinireddy/go/src/github.com/IBM-Cloud/terraform-provider-ibm/ibm/resource_ibm_iam_user_invite.go:513 +0x5fa
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0007cc6c0, 0x3b40de0, 0xc000538680, 0xc00042f680, 0x369aaa0, 0xc000bc4000, 0x0, 0x0, 0x0)
        /Users/harinireddy/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.4/helper/schema/resource.go:278 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007cc6c0, 0x3b40de0, 0xc000538680, 0xc0002ffd50, 0xc000b34640, 0x369aaa0, 0xc000bc4000, 0x0, 0x0, 0x0, ...)
        /Users/harinireddy/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.4/helper/schema/resource.go:396 +0x65b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000db80, 0x3b40de0, 0xc000538680, 0xc000aebb30, 0xc000538680, 0x0, 0x3b7bf80)
        /Users/harinireddy/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.4/helper/schema/grpc_provider.go:955 +0x8cf
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000cb7640, 0x3b40de0, 0xc000538680, 0xc0002ffab0, 0xc000cb7640, 0xc000b20810, 0xc000714ba0)
        /Users/harinireddy/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/server/server.go:332 +0xae
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x35c1500, 0xc000cb7640, 0x3b40ea0, 0xc000b20810, 0xc000f47860, 0x0, 0x3b40ea0, 0xc000b20810, 0xc000450600, 0x1fd)
        /Users/harinireddy/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a98380, 0x3b5d100, 0xc000093e00, 0xc000f66b00, 0xc000a59260, 0x4a2b720, 0x0, 0x0, 0x0)
        /Users/harinireddy/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x522
google.golang.org/grpc.(*Server).handleStream(0xc000a98380, 0x3b5d100, 0xc000093e00, 0xc000f66b00, 0x0)
        /Users/harinireddy/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd05
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000b6250, 0xc000a98380, 0x3b5d100, 0xc000093e00, 0xc000f66b00)
        /Users/harinireddy/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xa5
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /Users/harinireddy/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-ibm_v1.23.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply -var-file="secrets.tfvars"`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/resource_ibm_iam_user_invite.go
+++ b/ibm/resource_ibm_iam_user_invite.go
@@ -854,19 +854,22 @@ func getInfraPermissions(d *schema.ResourceData, meta interface{}) []string {
 	var infraPermissions = make([]string, 0)
 	if data, ok := d.GetOk("classic_infra_roles"); ok {
 		for _, resource := range data.([]interface{}) {
-			d := resource.(map[string]interface{})
-			if permissions, ok := d["permissions"]; ok && permissions != nil {
-				for _, value := range permissions.([]interface{}) {
-					infraPermissions = append(infraPermissions, fmt.Sprintf("%v", value))
+			if resource != nil {
+				d := resource.(map[string]interface{})
+				if permissions, ok := d["permissions"]; ok && permissions != nil {
+					for _, value := range permissions.([]interface{}) {
+						infraPermissions = append(infraPermissions, fmt.Sprintf("%v", value))
+					}
 				}
-			}
-			if permissionSet, ok := d["permission_set"]; ok && permissionSet != nil {
-				if permissions, ok := permissionSets[permissionSet.(string)]; ok {
-					for _, permission := range permissions {
-						infraPermissions = append(infraPermissions, permission)
+				if permissionSet, ok := d["permission_set"]; ok && permissionSet != nil {
+					if permissions, ok := permissionSets[permissionSet.(string)]; ok {
+						for _, permission := range permissions {
+							infraPermissions = append(infraPermissions, permission)
+						}
 					}
 				}
 			}
+
 		}
 		return infraPermissions
 	}

